### DISCUSSION TITLE
Can the bootstrapper be used for wrapping a .NET library for use in JS browser applications?

### DISCUSSION THREAD

**Discussion by petertiedemann** [unoplatform/Uno.Wasm.Bootstrap#507] [Q&A] [answered]
Hi,

I have a C# library that I would very much like to make available to existing JS applications. It would replace an HTTP request with in-browser computation, and could be wrapped in a static C# class with a single method accepting a request object and returning a response object (as objects, serialized as JSON if required). Basically enabling the app to switch between online and offline mode.

I am trying to find out if Uno.Wasm.Bootstrap is suitable for this purpose. I noticed that it explicitly requires an entry point, which does not naturally fit into this scenario. I can of course define an empty main method, but is there any support for explicitly "exporting" methods, or put differently: Can I be sure my API method will not be eliminated by tree shaking?

Is the way forward simply an empty main and a static method accessed using `Module.mono_bind_static_method`?

PS: I *am* aware of https://github.com/Elringus/DotNetJS which seems a better fit, but it does not support AOT which is critical for my use-case

**Comment by jeromelaban** [comment, marked as answer]
It is not yet possible, but .NET 7 is working to bringing what's required to this particularly scenario. .NET 6 was missing modularization, and it'll be brought forward then.

**Reply by petertiedemann** [reply]
> It is not yet possible, but .NET 7 is working to bringing what's required to this particularly scenario. .NET 6 was missing modularization, and it'll be brought forward then.

Thank you for the near instantaneous response :) However, I do not fully understand what is missing? You mention modularization, but essentially I am just looking to have a static method (like the entry-point), but only declared and not invoked by the generated JS file ( I assumed that I would have to use the ["BrowserEmbedded"](https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/main/doc/features-embedded.mode.md) mode. So naively I thought the JS app would just include the generated embedded.js file, and then have access to e.g. `Module.mono_bind_static_method`.

Based on the answer here I was hoping it might be possible: https://github.com/unoplatform/Uno.Wasm.Bootstrap/discussions/439. Just clarifying in case I gave the impression that I was expecting to include assemblies modules.

**Reply by jeromelaban** [reply]
You're welcome :)

The problem with .NET 6 is the JS globals that are registered when creating the runtime, something that will not happen with .NET 7. If there are globals, this means you can only have one instance of the runtime per page. If that's your scenario, then the current solution may be a good fit, otherwise, you'll need to wait for net7.

**Reply by petertiedemann** [reply]
Thank you for the clarification, a single instance of the runtime per page is perfectly fine in our scenario, but I can definitely see how it could be a significant limitation for general usage.

**Reply by petertiedemann** [reply]
Just FYI to anyone else reading this. I did have to ensure that the main method references the method I wanted to use from JS, otherwise it would be removed. @jeromelaban Do you happen to now if there is a nicer way to prevent trimming? (some attribute or similar?)

**Reply by jeromelaban** [reply]
For the IL Linker, see this: https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/main/doc/linker-configuration.md
